File: c++/1759.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int l, c;
vector<char> alphabets;
vector<char> code;

void dfs(int idx, int cnt) {
  if (cnt == l) {
    int vowelLength = 0;
    for (int i = 0; i < l; i++) {
      if (code[i] == 'a' || code[i] == 'e' || code[i] == 'i' ||
          code[i] == 'o' || code[i] == 'u') {
        vowelLength++;
      }
    }
    if (0 < vowelLength && vowelLength <= l - 2) {
      std::string str(code.begin(), code.end());
      cout << str << '\n';
    }
  }

  for (int i = idx; i < c; i++) {
    code.push_back(alphabets[i]);
    dfs(i + 1, cnt + 1);
    code.pop_back();
  }
}

int main() {
  freopen("example.txt", "r", stdin);

  cin >> l >> c;

  for (int i = 0; i < c; i++) {
    char c;
    cin >> c;
    alphabets.push_back(c);
  }
  sort(alphabets.begin(), alphabets.end());

  dfs(0, 0);
  return 0;
}

File: js/1759.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const [l, c] = inputs[0].split(' ').map(Number);
const alphabets = inputs[1].split(' ').sort();
const answer = [];
const code = [];

dfs(0, 0);
console.log(answer.join('\n'));

function dfs(idx, cnt) {
  if (cnt === l) {
    const vowelLength = code.filter(
      (a) => a === 'a' || a === 'e' || a === 'i' || a === 'o' || a === 'u'
    ).length;
    if (0 < vowelLength && vowelLength <= l - 2) {
      answer.push(code.join(''));
    }
  }

  for (let i = idx; i < c; i++) {
    code.push(alphabets[i]);
    dfs(i + 1, cnt + 1);
    code.pop();
  }
}
